tcf_website/views/auth.py

```diff
@@ -72,7 +72,12 @@ def collect_extra_info(request, method):
 
             # once we have the grad_year stashed in the session, we can
             # tell the pipeline to resume by using the "complete" endpoint
-            return redirect(reverse("social:complete", args=[method]))
+            return redirect(reverse("social:complete", args=[method])
+                            + "?verification_code="
+                            + request.GET['verification_code']
+                            + "&partial_token="
+                            + request.GET['partial_token']
```

Comment:
Previously, when the user set their graduation date, the auth pipeline would directly run `tcf_core.auth_pipeline.collect_extra_info`. Now that `tcf_core.auth_pipeline.collect_extra_info` uses the same partial token as `social_core.pipeline.mail.mail_validation`, when the user sets their graduation date the pipeline starts at `mail_validation`. The `verification_code` and `partial_token` must be in this redirect so that `mail_validation` finishes successfully.